text/dependencies/rfcs/0000-dependency-management-phase-two.md

```diff
@@ -0,0 +1,195 @@
+through a series of API endpoint routing.
+
+Dependencies are updated in the buildpacks in a workflow that runs
+in the buildpack's Github Actions:
+- **Update Dependencies**: Runs every hour and runs [`jam
+  update-dependencies`](https://github.com/paketo-buildpacks/github-config/blob/f85a65d14fd90f6d63f1af2d408ee38f17ce5c0b/actions/dependency/update/action.yml#L45-L46)
+  against the `buildpack.toml`. If there are new versions available in the
+  dep-server, this workflow will discover them and open a PR to update the
+  `buildpack.toml`.
+
+### New Plan Overview
+The new process for managing dependencies will use the `buildpack.toml` as a
+source of truth for the latest versions we support, instead of using a separate
+`known-versions.json` file. Metadata will also be generated and updated in
+place with a pull request to the `buildpack.toml`, rather than being pushed to
+an intermediary `metadata.json` file. The overall process will be closer to
+what someone would do if they were updating the dependencies themselves, rather
+than with automation. The overall steps will be:
+
+1. Pick up all versions of the dependency, using code from the Phase 1 RFC in
+   `<buildpack>/dependency/retrieval`
+2. Compare the list of all versions with the version constraints and supported
+  versions in the `buildpack.toml` to determine what versions can be updated.
+3. Generate metadata for each of the versions to be added using code from Phase
+   1 in `<buildpack>/dependency/metadata`.
+4. If the `URI` and `SHA256` are missing from the metadata, the code will be
+  compiled with the code from `<buildpack>/dependency/compilation`.
+5. Test the dependency (whether compiled or not) using the test from
+   `<buildpack>/dependency/test`
+6. (If compiled) Upload the dependency to the dependency GCP bucket
+7. (If compiled) Add the GCP bucket access URI and the SHA256 of the dependency to the
+    metadta `URI` and `SHA256` fields
+8. Update the `buildpack.toml` with the new versions and metadata
+
+This new plan eliminates extra steps of storing metadata in a file for later
+use, and then separately updating the dependencies on a timer.
+
+### Workflows and Actions
+Pending the Phase 1 RFC is accepted, and it's decided dependency logic will
+live alongside the buildpack, workflows will be rolled out to all of the
+buildpacks from
+[paketo-buildpacks/github-config](https://github.com/paketo-buildpacks/github-config)
+inside the `implementation/.github/workflows` location.
+
+Actions will live in the github-config repository as well, under a
+directory named `dependency`, inside of the `actions` directory.
+
+The github-config repository `CODEOWNERS` file will be updated to set the
+`@paketo-buildpacks/dependencies-maintainers` to the owners of the
+`actions/dependency` and `implementation/.github/workflows/<all
+dependency-related workflows>`.
+
+
+Each of the step s outlined in the overview will be converted into workflow
+steps.
+
+Workflow 1:
```

Comment:
I'd find it easier to understand the explanation of workflows if they were pseudo-coded as actual workflow yaml, maybe with explanatory comments sprinkled throughout.

Reply:
I think that explaining the workflows this way is easier for someone less familiar 
with Github Actions details to get the point across of what the automation is responsible for, which is all I wanted for this section. This avoids shoehorning the details of the workflows, which may change when actually implemented. I think I'd prefer to leave the explanation at this level for the RFC, especially since the workflows will be implemented once and rolled out to buildpacks, rather than buildpacks implementing their own.